server/apps/users/services.py

```diff
@@ -102,6 +103,29 @@ def compare_preferences(original, updates):
     return added, removed, modified
 
 
+def set_sign_off(user):
+    """
+    Set sign_off property on user if it's not set already.
+    """
+
+    if SIGN_OFF not in user:
+        if 'first_name' not in user or 'last_name' not in user:
+            user[SIGN_OFF] = user['username'][:3].upper()
+        else:
+            user[SIGN_OFF] = '%s%s' % (user['first_name'][0], user['last_name'][0])
+
+
+def get_sign_off(user):
+    """
+    Gets sign_off property on user if it's not set already.
+    """
+
+    if SIGN_OFF not in user:
+        set_sign_off(user)
+
+    return user[SIGN_OFF]
```

Comment:
Does `set_sign_off()` set the `SIGN_OFF` key?
(if not, this might throw a `KeyError`)